core.cc

```diff
@@ -0,0 +1,152 @@
+#include <node.h>
+#include <nan.h>
+#include <syslog.h>
+
+using v8::Function;
+using v8::FunctionTemplate;
+using v8::Handle;
+using v8::Local;
+using v8::Null;
+using v8::Number;
+using v8::Object;
+using v8::String;
+using v8::Value;
+
+class Worker : public NanAsyncWorker {
+    public:
+        Worker(NanCallback *callback, int priority, Handle<String> message_)
+            : NanAsyncWorker(callback), priority(priority), message(0) {
+                message = new char[message_->Length() * 4];
+                message_->WriteUtf8(message);
```

Comment:
You can check with `node::Buffer::HasInstance()` whether a `Handle<Value>` is a Buffer or not:

```c++
#include <node_buffer.h>

// ...

if (node::Buffer::HasInstance(message_)) {
  const char* data = node::Buffer::Data(message_);
  const size_t length = node::Buffer::Length(message_);
  message = new char[length];
  memcpy(message, data, length);
} else {
  // ...
}
```

PS: If you want to future-proof your code, use `Local<T>`.  `Handle<T>` will be removed from V8 someday.